src/celeritas/optical/gen/GeneratorData.hh

```diff
@@ -152,6 +155,14 @@ struct GeneratorState : public GeneratorStateBase
 {
     CollectionStateStore<GeneratorStateData, M> store;
 
+    //! Access valid range of distributions
+    auto distributions()
+    {
+        return this->store.ref()
+            .distributions[ItemRange<GeneratorDistributionData>(
+                ItemId<GeneratorDistributionData>(this->counters.buffer_size))];
```

Comment:
A comment about the return type would be helpful: I guess it's auto because it's a span for host data, an LdgSpan for device?

Reply:
Mea culpa 😅